modules/network/samba.c

```diff
@@ -51,33 +51,36 @@ void
 scan_samba_usershares(void)
 {
     FILE *usershare_list;
-    
-    if ((usershare_list = popen("net usershare list", "r"))) {
-        char buffer[512];
-        
+    gchar *out, *err;
+    gboolean spawned;
+    int status;
+    gchar *p, *next_nl;
+    gchar *usershare, *cmdline;
+    gsize length;
+
+    spawned = g_spawn_command_line_sync("net usershare list",
+            &out, &err, &status, NULL);
```

Comment:
If it's leaking, it's a bug in GLib. It's marked as [optional](https://developer.gnome.org/glib/stable/glib-Spawning-Processes.html#g-spawn-command-line-sync) in the documentation.